docusaurus/docs/React/guides/theming/input-ui.mdx

```diff
@@ -3,213 +3,810 @@ id: input_ui
-to customize styling is to write your own CSS and override the default values. To get a sense of all the classes and styles
-applied to the `MessageInput` and its child components, you can either inspect the DOM inside your browser.
+Message input is a bit more complex than it might seem at first glance. Not just a text box with a
+"send" button, it has a lot of hidden features:
 
-For example, here's how you can override the styles for the underlying `textarea` and make its background light blue:
+1. Updating the typing status
+2. Uploading and previewing attachments
+3. Displaying link previews
+4. Auto-completing mentions, commands, emoji...
+
+![](../../assets/message-input-ui-kitchensink.png)
+
+We will cover this features step by step. For now, let's start with the simplest markup possible:
+
+<Tabs groupId="example">
+<TabItem value="js" label="React">
+
+```jsx
+function CustomMessageInput() {
+  return (
+    <div className='message-input'>
+      <textarea value={text} className='message-input__input' />
+      <button type='button' className='message-input__button'>
+        ⬆️
+      </button>
+    </div>
+  );
+}
+```
+
+</TabItem>
+<TabItem value="css" label="CSS">
 
 ```css
-.str-chat .str-chat__textarea > textarea {
-  background-color: #99ccff;
+.message-input {
+  display: flex;
+  align-items: center;
+  border: 2px solid #00000029;
+  border-radius: 8px;
+}
+
+.message-input:has(.message-input__input:focus) {
+  border-color: #005fff;
+}
+
+.message-input__input {
+  flex-grow: 1;
+  border: 0;
+  outline: 0;
+  background: none;
+  font: inherit;
+  padding: 8px;
+  resize: none;
+}
+
+.message-input__button {
+  border: 1px solid transparent;
+  outline: 0;
+  background: none;
+  font: inherit;
+  border-radius: 4px;
+  margin: 8px;
+  padding: 8px;
+  cursor: pointer;
+}
+
+.message-input__button:hover {
+  background: #fafafa;
+  border-color: #00000014;
+}
+
+.message-input__button:focus {
+  border-color: #005fff;
 }
 ```
 
-### Customize Functionality
+</TabItem>
+</Tabs>
 
-The `MessageInput` component supports a variety of props that let you customize its behavior in various ways.
-You can utilize props to change basic behaviors of the underlying `textarea` element (ex: the `grow` prop lets you specify
-whether the input field should automatically increase in height when the message wraps the input), as well as override
-functions and more complex and logic-based parts of the component (ex: the `doImageUploadRequest` prop lets you supply a custom
-function that handles the uploading of image attachments).
+![](../../assets/message-input-ui-simple.png)
 
-:::note
-For a complete overview of props, take a look at the [MessageInput props](../../components/message-input-components/message-input.mdx#props) section.
-:::
+Note that you should not render your custom message input directly, but instead pass it as a prop to
+either `Channel` or `MessageInput` component. That way, you can be sure that your input is wrapped
```

Comment:
How do you feel about adding links to the components that are being mentioned? I'm adding links to the first occurences within the guide but now thinking about it I realize that sometimes we direct readers to specific chapters which mention components that have no links because it's not their first occurence within the guide. All the links or no links at all?

Reply:
This is a good point, and my vote is for ALL the links.

Reply:
Added ALL the links :)